### DISCUSSION TITLE
Using Prisma2 to auto-generate resolvers. Now front-end requirements does not exactly match with what the resolvers are providing. Is there a way to extend, override the resolvers that are being generated?

### DISCUSSION THREAD

**Discussion by palashCItobuz** [MichalLytek/typegraphql-prisma#173] [Q&A]
```
 async deleteManyEvent(@TypeGraphQL.Ctx() ctx: any, @TypeGraphQL.Info() info: GraphQLResolveInfo, @TypeGraphQL.Args() args: DeleteManyEventArgs): Promise<AffectedRowsOutput> {
    const { _count } = transformFields(
      graphqlFields(info as any)
    );
    return getPrismaFromContext(ctx).event.deleteMany({
      ...args,
      ...(_count && transformCountFieldIntoSelectRelationsCount(_count)),
    });
  }
  ```
For example, here i need to soft-delete multiple events at once and add the deleteDate in the deletedAt field. How do I achieve this and what is the ideal way? 
Another readAt field is required to be updated when the user fetches a single event. For that that field needs to be updated while querying the event for the first time.

**Comment by MichalLytek** [comment]
You can pick only the generated resolvers you want and then write custom resolvers with custom logic.
There's no way to override or extend the registered generated resolver.

**Reply by rmrantunes** [reply]
I am pretty interested in how possible is to do this. For example, every time a Post is created I need to connect to an external service like Firebase Cloud Messaging to send notifications to subscribed users. Is it possible or am I too newbie to GraphQL?

**Comment by rmrantunes** [comment]
Actually I have achieved an acceptable result by just creating a new TypeGraphQL resolver with same name (createPost) and concat to the array of resolvers, so it can be overwritten

```ts
// resolvers/Post.ts

import { PrismaClient } from '.prisma/client'
import { PostCreateInput, Post } from '@generated/type-graphql'
import { Arg, Ctx, Mutation, Resolver } from 'type-graphql'

@Resolver()
export class MyPostCrudResolver {
  @Mutation(() => Post)
  async createPost(
    @Arg('data') data: PostCreateInput,
    @Ctx() ctx: { prisma: PrismaClient }
  ): Promise<Post> {
    const post = await ctx.prisma.post.create({ data })

    // my custom logic

    return post
  }
}


```

```ts 
// resolvers/index.ts

import { resolvers as generatedResolvers } from '@generated/type-graphql'
import { NonEmptyArray } from 'type-graphql'
import { MyPostCrudResolver } from './Post'

export const resolvers = generatedResolvers
  .concat(MyPostCrudResolver) as NonEmptyArray<Function>
```

And so I use the resolvers exported above instead of those from `'@generated/type-graphql'`
